client/elasticsearch/v710/client_test.go

```diff
@@ -39,3 +73,84 @@ func TestMultiSearch(t *testing.T) {
 		So(splitQuery[3], ShouldEqual, "{\"query\" : {\"match_all\" : {}}}")
 	})
 }
+
+func TestDeleteDocument(t *testing.T) {
+	Convey("Given a valid ESClient", t, func() {
+		esClient := newMockClient(http.StatusOK, `{}`, nil)
+		testClient := &ESClient{esClient: esClient}
+
+		Convey("When DeleteDocument returns 200", func() {
+			err := testClient.DeleteDocument(context.Background(), "my-index", "my-id")
+			So(err, ShouldBeNil)
+		})
+
+		Convey("When DeleteDocument returns 500", func() {
+			esClient := newMockClient(http.StatusInternalServerError, `{"error":"server error"}`, nil)
+			testClient := &ESClient{esClient: esClient}
+			err := testClient.DeleteDocument(context.Background(), "my-index", "my-id")
+			So(err, ShouldNotBeNil)
+			So(err.Error(), ShouldContainSubstring, "delete request failed")
+		})
+	})
+}
+
+func TestDeleteDocumentByQuery(t *testing.T) {
+	Convey("Given a valid ESClient", t, func() {
+		var receivedBody string
+		assertBody := func(req *http.Request) {
+			bodyBytes, _ := io.ReadAll(req.Body)
+			receivedBody = string(bodyBytes)
+		}
+
+		esClient := newMockClient(http.StatusOK, `{}`, assertBody)
+		testClient := &ESClient{esClient: esClient}
+
+		Convey("When DeleteDocumentByQuery sends a correct query", func() {
+			queryMap := map[string]interface{}{
+				"query": map[string]interface{}{
+					"term": map[string]interface{}{
+						"uri": "/my/uri",
+					},
+				},
+			}
+
+			queryBytes, err := json.Marshal(queryMap)
+			So(err, ShouldBeNil)
```

Comment:
I wouldn't bother marshalling a map, just use the json directly
```suggestion
			queryBytes := []byte(`{"query": {"term": {"uri": "/my/uri"}}}`)
```

Reply:
Also same for example in next test too.